cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -1257,6 +1282,64 @@ def make_resource(self, data, **kwargs):
         return HeadNode(**data)
 
 
+class LoginNodeImageSchema(BaseSchema):
+    custom_ami = fields.Str(validate=validate.Regexp(PCLUSTER_AMI_ID_REGEX))
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return LoginNodeImage(**data)
+
+
+class LoginNodeSshSchema(BaseSchema):
+    key_name = fields.Str(required=True)
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return LoginNodeSsh(**data)
+
+
+class LoginNodeNetworkingSchema(BaseNetworkingSchema):
+    subnet_id = fields.Str(required=True)
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return LoginNodeNetworking(**data)
+
+
+class LoginNodePoolSchema(BaseSchema):
+    name = fields.Str(required=True)
+    instance_type = fields.Str(required=True)
+    image = fields.Nested(LoginNodeImageSchema)
+    networking = fields.Nested(LoginNodeNetworkingSchema, required=True)
+    count = fields.Int(required=True, validate=validate.Range(min=1))
+    ssh = fields.Nested(LoginNodeSshSchema, required=True)
+    iam = fields.Nested(LoginNodeIamSchema)
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return LoginNodePool(**data)
+
+
+class LoginNodesSchema(BaseSchema):
+    pools = fields.Nested(
+        LoginNodePoolSchema,
+        many=True,
+        required=True,
+        validate=validate.Length(min=1, max=1, error="has a minimum size of 1, "
```

Comment:
The error message will probably be revised later but for now, let's go with a tentative message: "Only one pool can be specified when using login nodes."

Reply:
Also, you can use equal instead of setting a min and max of 1 ([Length](https://marshmallow.readthedocs.io/en/stable/marshmallow.validate.html#marshmallow.validate.Length))

Reply:
Done!